src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,205 @@
+      "circumscribes the cube containing both compact objects, and the \n"
+      "radius of the outer boundary. The options ExciseInteriorA and \n"
+      "ExciseInteriorB determine whether the layer-zero blocks are present \n"
+      "inside each compact object. If set to `true`, the domain will not \n"
+      "contain layer zero for that object. The user specifies XCoordObjectA \n"
+      "and XCoordObjectB, the x-coordinates of the locations of the centers \n"
+      "of each compact object. In these coordinates, the location for the \n"
+      "axis of rotation is x=0. Please make sure that your choices of \n"
+      "locations along with the resulting center of mass is compatible with \n"
+      "this."};
+
+  BinaryCompactObject(
+      typename InnerRadiusObjectA::type inner_radius_objectA,
+      typename OuterRadiusObjectA::type outer_radius_objectA,
+      typename XCoordObjectA::type xcoord_objectA,
+      typename ExciseInteriorA::type excise_interiorA,
+      typename InnerRadiusObjectB::type inner_radius_objectB,
+      typename OuterRadiusObjectB::type outer_radius_objectB,
+      typename XCoordObjectB::type xcoord_objectB,
+      typename ExciseInteriorB::type excise_interiorB,
+      typename RadiusOuterCube::type radius_enveloping_cube,
+      typename RadiusOuterSphere::type radius_enveloping_sphere,
+      typename InitialRefinement::type initial_refinement,
+      typename InitialGridPoints::type initial_number_of_grid_points,
+      typename UseEquiangularMap::type use_equiangular_map,
+      typename UseProjectiveMap::type use_projective_map = true,
+      const OptionContext& context = {});
+
+  BinaryCompactObject() = default;
+  BinaryCompactObject(const BinaryCompactObject&) = delete;
+  BinaryCompactObject(BinaryCompactObject&&) noexcept = default;
+  BinaryCompactObject& operator=(const BinaryCompactObject&) = delete;
+  BinaryCompactObject& operator=(BinaryCompactObject&&) noexcept = default;
+  ~BinaryCompactObject() noexcept override = default;
+
+  Domain<3> create_domain() const noexcept override;
+
+  std::vector<std::array<size_t, 3>> initial_extents() const noexcept override;
+
+  std::vector<std::array<size_t, 3>> initial_refinement_levels() const
+      noexcept override;
+
+ private:
+  typename InnerRadiusObjectA::type inner_radius_objectA_{};
+  typename OuterRadiusObjectA::type outer_radius_objectA_{};
+  typename XCoordObjectA::type xcoord_objectA_{};
+  typename ExciseInteriorA::type excise_interiorA_{};
+  typename InnerRadiusObjectB::type inner_radius_objectB_{};
+  typename OuterRadiusObjectB::type outer_radius_objectB_{};
+  typename XCoordObjectB::type xcoord_objectB_{};
+  typename ExciseInteriorB::type excise_interiorB_{};
+  typename RadiusOuterCube::type radius_enveloping_cube_{};
+  typename RadiusOuterSphere::type radius_enveloping_sphere_{};
+  typename InitialRefinement::type initial_refinement_{};
+  typename InitialGridPoints::type initial_number_of_grid_points_{};
```

Comment:
I suggest renaming `initial_number_of_grid_points_` to `initial_grid_points_per_dim_`.

I suggest this because the function `Mesh::number_of_grid_points()` returns the total number of grid points, and it could be misleading to have this variable with such a similar name when it encodes a different quantity.